server/src/services/asset.service.ts

```diff
@@ -249,7 +249,18 @@ export class AssetService extends BaseService {
 
     const { thumbnailFile, previewFile } = getAssetFiles(asset.files);
     const files = [thumbnailFile?.path, previewFile?.path, asset.encodedVideoPath];
-    if (deleteOnDisk) {
+
+    let willDelete = deleteOnDisk;
+
+    if (asset.isOffline) {
+      /* We don't want to delete an offline asset because it is either...
+           ...missing from disk => don't delete the file since it doesn't exist where we expect
+           ...outside of any import path => don't delete the file since we're not responsible for it
+           ...matching an exclusion pattern => don't delete the file since we're not responsible for it */
+      willDelete = false;
+    }
```

Comment:
This logic already exists outside of this job. Why is it duplicated here again?

Reply:
Did you only keep the logic inside here? Imo the job should be called with delete: true | false and then that's what should happen. It seems much clearer to just pass false when the asset is offline.

Reply:
My reasoning was that it would be safer for the future to check the isOffline right before deletion. The possible outcome is data loss which is severe and I didn't want future refactors to accidently remove the check. But I see the logic in your approach and will update accordingly.

Reply:
When an asset goes online the offline flash and the trashed flag are updated at the same time and vice versa. If a file goes offline after the job is queued I don't see any reason with still trying to delete it from disk. If a file went online after the job was queued, it queued it with `delete: false` anyways.

Reply:
> When an asset goes online the offline flash and the trashed flag are updated at the same time and vice versa. If a file goes offline after the job is queued I don't see any reason with still trying to delete it from disk. If a file went online after the job was queued, it queued it with `delete: false` anyways.

Sounds good!